research/validation/phase0a_emission_entropy.py

```
@@ -0,0 +1,362 @@
+                reasoning = extract_reasoning(response)
+
+                if tokens is not None:
+                    all_tokens.append(tokens)
+                    print(f"OK ({len(tokens)} tokens)", end="")
+                else:
+                    parse_failures += 1
+                    print("PARSE_FAIL", end="")
+
+                if reasoning is not None:
+                    reasoning_present += 1
+                    print(f" +reasoning", end="")
```

Ruff (0.15.14)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)